### PR TITLE
Enhancement/issue 8427

### PR DESCRIPTION
**Pull request by elsayed2440**
### Describe your change:
Impelementation of 2 algorithms from the ones mentioned in the issue #8427 , namely the sliding window algorithm and Kadane's algorithm.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/max_sub_array_sum.py
Comment (tianyizheng02) [inline dynamic_programming/max_sub_array_sum.py:1]: Unfortunately we already have an implementation of max subarray sum in [`maths/kadanes.py`](https://github.com/TheAlgorithms/Python/blob/master/maths/kadanes.py), [`other/maximum_subarray.py`](https://github.com/TheAlgorithms/Python/blob/master/other/maximum_subarray.py), [`maths/largest_subarray_sum.py`](https://github.com/TheAlgorithms/Python/blob/master/maths/largest_subarray_sum.py), [`other/maximum_subsequence.py`](https://github.com/TheAlgorithms/Python/blob/master/other/maximum_subsequence.py), and [`dynamic_programming/max_sum_contiguous_subsequence.py`](https://github.com/TheAlgorithms/Python/blob/master/dynamic_programming/max_sum_contiguous_subsequence.py). Furthermore, [`dynamic_programming/max_sub_array.py`](https://github.com/TheAlgorithms/Python/blob/master/dynamic_programming/max_sub_array.py) and [`divide_and_conquer/max_subarray_sum.py`](https://github.com/TheAlgorithms/Python/blob/master/divide_and_conquer/max_subarray_sum.py) both implement this algorithm has a subroutine.

... I have no idea why there are so many near-duplicate implementations of the same algorithm, but this _really_ shouldn't be the case. I'll open an issue to address this.

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: dynamic_programming/max_sub_array_sum.py
Comment (tianyizheng02) [reply dynamic_programming/max_sub_array_sum.py:1]: Opened #8812

**Comment by tianyizheng02** [comment]
Also, some of these commits don't pertain to the algorithm contribution itself (Create/Delete TestShiva, Add black version/delete black file, etc.), so this PR could use a rebase to get those commits deleted/squashed before it's merged

File: web_programming/sliding_window_generators.py
[the remark this answers is not in the document]
Comment (elsayed2440) [reply web_programming/sliding_window_generators.py:1]: Thank you for the clarification, I have moved the files to other folder

File: web_programming/sliding_window.py
[the remark this answers is not in the document]
Comment (elsayed2440) [reply web_programming/sliding_window.py:1]: Thanks, I have modified and filled it in

**Comment by cclauss** [comment]
In the commit body, the box is checked next to
>  This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.

yet there are three algorithms in this file.

How is the first file better than
https://github.com/TheAlgorithms/Python/blob/fa12b9a286bf42d250b30a772e8f226dc14953f4/maths/largest_subarray_sum.py#L4

File: dynamic_programming/max_sub_array_sum.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        arr (list): The input array of numbers.
+        size (int): The size of the array.
+
+    Returns:
+        int: The maximum sum of a subarray within the array.
+
+    Example:
+        >>> arr = [-2, -3, 4, -1, -2, 5, -3]
+        >>> max_sub_array_sum(arr, len(arr))
+        6
```
Comment (cclauss) [inline dynamic_programming/max_sub_array_sum.py:15]: Add a blank line to the tests pass.
```suggestion
        6

```
As discussed in `CONTRIBUTING.md` please test your code by running on your local machine:
% `python3 -m doctest -v dynamic_programming/max_sub_array_sum.py`

File: other/sliding_window.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    https://stackoverflow.com/questions/8269916/what-is-sliding-window-algorithm-examples
+    """
+
+    if len(elements) <= window_size:
+        return [elements]
+
+    windows = []
+    for i in range(len(elements) - window_size + 1):
+        window = elements[i : i + window_size]
+        windows.append(window)
```
Comment (cclauss) [inline other/sliding_window.py:28]:
```suggestion
        windows.append(elements[i : i + window_size])
```

File: other/sliding_window_generators.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    Returns:
+        generator: A generator that yields sublists of size window_size.
+
+    Example:
+        >>> lst = [1, 2, 3, 4, 5, 6, 7, 8]
+        >>> sw_gen = sliding_window(lst, 3)
+        >>> print(next(sw_gen))
+        [1, 2, 3]
+        >>> print(next(sw_gen))
```
Comment (cclauss) [inline other/sliding_window_generators.py:22]:
```suggestion
        >>> next(sw_gen)
        [1, 2, 3]
        >>> next(sw_gen)
```

**Pull request closed without merging**